leases/lease.go

```diff
@@ -65,6 +65,12 @@ func SynchronousDelete(ctx context.Context, o *DeleteOptions) error {
 	return nil
 }
 
+// WithLabel sets a label on a lease, and merges it with existing labels.
+// It overwrites the existing value of the given label (if present).
+func WithLabel(label, value string) Opt {
+	return WithLabels(map[string]string{label: value})
```

Comment:
nit: We could avoid the extra map allocation by inlining (and rewriting) the `WithLabels`:

```suggestion
	return func(l *Lease) error {
		if l.Labels == nil {
			l.Labels = map[string]string{label: value}
			return
		}
		l.Labels[label] = value
		return nil
}
```

Reply:
Ah, yes, that's better. Didn't think about the extra allocation here, but makes sense; let me update 👍

Reply:
Updated 👍